packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt

```diff
@@ -110,6 +174,188 @@ class FileDownloader(
     }
   }
 
+  private fun isPatchResponse(responseBody: ResponseBody): Boolean {
+    val mediaType: MediaType? = responseBody.contentType()
+    return mediaType?.let { it.type == "application" && it.subtype == "vnd.bsdiff" } ?: false
+  }
+
+  @Throws(IOException::class)
+  private fun applyPatchAndVerify(
+    asset: AssetEntity,
+    request: Request,
+    responseBody: ResponseBody,
+    destination: File,
+    updatesDirectory: File,
+    expectedBase64URLEncodedSHA256Hash: String?
+  ): ByteArray {
+    val launchAssetContext = prepareAssetForDiff(
+      asset,
+      request,
+      responseBody,
+      updatesDirectory
+    )
+
+    val requestedUpdateId = request.header(EXPO_REQUESTED_UPDATE_ID_HEADER)
```

Comment:
I don't think these headers are in the protocol spec: https://docs.expo.dev/technical-specs/expo-updates-1/. That's not necessarily an issue though since the boundary of what the protocol specifies could be higher up, or potentially this isn't in the protocol whatsoever? (ability to request/apply diff is implementation-specific)

Reply:
Actually the protocol is written more loosely than our implementation. It says the library should include the content type specified in the manifest? https://docs.expo.dev/technical-specs/expo-updates-1/#asset-request